views/js/pciCreator/dev/audioRecordingInteraction/runtime/js/recorder.js

```diff
@@ -107,6 +118,34 @@ define([
             analyser,               // the WebAudio node use to read the input level
             frequencyArray;         // used to compute the input level from the current array of frequencies
 
+        /**
+         * Check if client uses the iOS device.
+         *
+         * @param {Array} frequencyArray
+         * @returns {*|boolean}
+         */
+        function isIOSDevice(frequencyArray) {
+            return frequencyArray.length && /(iPhone|iPad)/i.test(navigator.userAgent)
```

Comment:
Is it the right place to check length of `frequencyArray`?
This function should only take care of the OS check.

Reply:
Moved the length check to the place of `isOSDevice` invocation.